### DISCUSSION TITLE
What exactly replaces resolve_xxx(self) in V1 ?

### DISCUSSION THREAD

**Discussion by GTorreil** [vitalik/django-ninja#937] [Q&A]
Hey there, 
Congratulations for releasing V1 :) 

I'm migrating my app to it, but there's something I don't really understand. 
The [documentation here](https://django-ninja.dev/whatsnew_v1/) says : 

> resolve_xxx(self, ...) - support resolve with (self) is dropped in favor of pydantic build-in functionality

What is the pydantic functionnality that replaces it ?

**Comment by vitalik** [comment]
Hi @GTorreil 

it's ["Computed Field"](https://docs.pydantic.dev/2.0/usage/computed_fields/)

```Python
from pydantic import BaseModel, computed_field


class Rectangle(BaseModel):
    width: int
    length: int

    @computed_field
    @property
    def area(self) -> int:
        return self.width * self.length


print(Rectangle(width=3, length=2).model_dump())
#> {'width': 3, 'length': 2, 'area': 6}
```

**Reply by GTorreil** [reply]
I see, but how do I access the ORM instance in the computed field ? I have an instance with annotations for M2M fields (using django's Prefetch object) and I need to access the annotation in the computed property to return a `list[int]`.

**Reply by GTorreil** [reply]
Hey @vitalik , sorry for the ping, but I still can't find how to compute fields based on arbitrary properties of the ORM instance. For instance, being able to populate a schema field with `[user.id for user in instance.users]` .
I hope this is clear ! cheers

**Reply by vitalik** [reply]
I would use the resolve (static) then:


```Python

class Some(Schema):
    id: int
    user_ids: list[int]

    @staticmethod
    def resolve_user_ids(instance):
         return [user.id for user in instance.users]
```

**Reply by yshalsager** [reply]
I am having the same problem and the suggestion @vitalik provided doesn't work for me unfortunately, my use case requires accessing a private field inside resolver like this:

```python
class Folder(Schema):
    _app = sys.modules[get_app_name(__package__)]

    id: int
	…
    folders: list[dict[str, Any]] | None = None
    
    def resolve_folders(self, folder, context):
        if 'folders' not in get_include_query(context):
            return None
        request = context['request']
        # access self._app here for some checks
```

Is there a better/recommended way rather than non-static resovler?

**Reply by GTorreil** [reply]
@yshalsager , resolvers are static methods. AFAIK they do not access the schema instance. The argument they get is the instance that was passed to `from_orm`, not the schema instance itself.

For this usecase, Pydantic computed fields should work.

**Reply by yshalsager** [reply]
> For this usecase, Pydantic computed fields should work. 

@GTorreil cirrect, but computed fields can't access object or context, which I need to access beside the private field.
